flycheck.el

```diff
@@ -6248,6 +6249,66 @@ Requires DMD 2.066 or newer.  See URL `http://dlang.org/'."
          (one-or-more " ") (message) line-end))
   :modes d-mode)
 
+(defun flycheck-elixir--find-default-directory (checker)
+  "Come up with a suitable default directory to run CHECKER in.
+
+This will either be the directory that contains `mix.exs' or,
+if no such file is found in the directory hierarchy, ths directory
+of the current file."
+  (or
+   (locate-dominating-file (buffer-file-name) "mix.exs")
+   (file-name-directory (buffer-file-name))))
```

Comment:
Is there a reason why you didn't choose the standard `default-directory` as fallback here?